decision/engine.go

```diff
@@ -115,29 +96,19 @@ type Engine struct {
 }
 
 // NewEngine creates a new block sending engine for the given block store
-func NewEngine(ctx context.Context, bs bstore.Blockstore, peerTagger PeerTagger) *Engine {
```

Comment:
Can we keep the peer tagger? That was a pretty important feature.

Reply:
I know. It'd need a more custom fix-- the old peer request queue doesn't exactly support it. gimme a bit if you want me to put it back in.

The other option is I can try to track down the performance problem in the PeerTaskQueue. What's our go for releasing 0.4.22?

Reply:
Looks related to freezing. Let's try to track this down, I think that'll be faster.